### ISSUE TITLE
End user license agreement for click-through access to media segments

### ISSUE THREAD

**Issue by mbergman42** [cta-wave/Test-Content-Generation#38]
The Tears of Steel and Big Buck Bunny Creative Commons Attribution licenses apply to WAVE test media files, but CTA Legal required some additional terms.  These additional terms were intended to be in an end-user license agreement (EULA), which should be presented to users when they seek to access WAVE test content.  (This is only for the test media, not the test software.)  On clicking "I accept", the users would gain normal access to the test content.  Currently we use github to host test content and there is no click-through agreement.

**Comment by mbergman42** [comment]
Text of the EULA would be,

### Beginning of EULA

**WAVE TEST MATERIAL USER AGREEMENT**

This user agreement (the “License ”) is entered into and is effective as of the date of content download between you (“Test Material User”) and the Consumer Technology Association (CTA)® located at 1919 South Eads Street, Arlington, VA 22202.

**WHEREAS**, CTA Web Application Video Ecosystem (“WAVE”) is an industry project on voluntary consensus technical standards and best practices for commercial internet video interoperability, organized and sponsored by CTA. The WAVE project provides encoded, digital entertainment audio and video content which may be used for testing purposes to verify conformance to WAVE Standards (“Test Material”).

**TERMS:**

1.  _License_. In consideration of the terms set forth below, CTA grants Test Material User license to use the Test Material for WAVE testing purposes subject to Creative Commons Attribution 4.0 Public Licensure (“CCA 4.0”).  Test Material User receives no patent, trademark or other similar rights in the Test Material under this License.

2.  _Media Format_. The license granted includes the right to change the media format of the Test Material.  

3.  _No Additional Restrictions_. Test Material User may not apply legal terms or technological measures to the Test Material or any derivative works that legally or practically restrict other users from doing anything that would otherwise be permitted if they licensed the Test Material directly from CTA. 

4.	_Ownership_. For purposes of this public license, Test Material User shall make no claim as to any ownership interest in the Test Materials or any portion thereof.

5.  _No Endorsement_.  Nothing in this License constitutes or may be construed as permission to assert or imply that Test Material Users are, or that Test Material Users’ use of the Test Material is, connected with, or sponsored, endorsed, or granted official status by, CTA or others designated to receive attribution.

6. _Attribution_.  When using the Test Material, Test Material Users must comply with CCA 4.0 Section 3. Test Material User must not in any way suggest that CTA endorses Test Material User or its use of the Test Material.

7.  _Disclaimer of Warranties/Limitation of Liability_.

a.	CTA offers the Test Material as-is and as-available and makes no representations or warranties of any kind concerning the Test Material, whether express, implied, statutory, or other. This includes, without limitation, warranties of title, merchantability, fitness for a particular purpose, non-infringement, absence of latent or other defects, accuracy, or the presence or absence of errors, whether or not known or discoverable. 

b.	To the extent possible, in no event will CTA be liable to Test Material User on any legal theory (including, without limitation, negligence) or otherwise for any direct, special, indirect, incidental, consequential, punitive, exemplary, or other losses, costs, expenses, or damages arising out of this License or use of the Test Material, even if CTA has been advised of the possibility of such losses, costs, expenses, or damages. 

c.	The disclaimer of warranties and limitation of liability provided above shall be interpreted in a manner that, to the extent possible, most closely approximates an absolute disclaimer and waiver of all liability.

d.	Indemnity.  Test Material User shall indemnify and hold CTA harmless against and from any third-party claim resulting from Test Material User’s use of the Test Materials. 

e.	IN NO EVENT SHALL CTA BE LIABLE TO COMPANY FOR AN AMOUNT IN EXCESS OF THE AMOUNT OF FEES PAID BY COMPANY UNDER THIS AGREEMENT FOR THE CALENDAR YEAR IN QUESTION BASED ON ANY CLAIM ARISING OUT OF OR RELATED TO THIS AGREEMENT, REGARDLESS OF WHETHER THE CLAIM IS BASED ON WARRANTY, CONTRACT, TORT OR ANY OTHER LEGAL THEORY.

8.  _Other Terms and Conditions_.

a.	CTA shall not be bound by any additional or different terms or conditions communicated by Test Material User unless expressly agreed in writing.
b.	CCA 4.0 as integrated into this License contains the complete and exclusive statement of the agreement between the parties and supersedes all prior agreements and understandings (whether oral or written) between the parties.  This License may not be modified or amended except in a writing executed by both parties that refers to this License.  

9.  _Interpretation_.

a.	To the extent possible, if any provision of this License is deemed unenforceable, it shall be automatically reformed to the minimum extent necessary to make it enforceable. If the provision cannot be reformed, it shall be severed from this License without affecting the enforceability of the remaining terms and conditions.

b.	No term or condition of this License will be waived and no failure to comply consented to unless expressly agreed to by CTA in writing.

c.	Nothing in this License constitutes or may be interpreted as a limitation upon, or waiver of, any privileges and immunities that apply to CTA or Test Material User, including from the legal processes of any jurisdiction or authority.

10.  _Governing Law_. This Agreement and all its rights and obligations of the parties relating hereto will be governed by and construed in accordance with the internal laws of the Commonwealth of Virginia without giving effect to any conflicts of law rules that would cause the application of the laws of any other jurisdiction. Any controversy or claim arising out of or relating to this Agreement or the breach thereof (except for an action for injunctive relief) may be resolved by arbitration held in Arlington County, Commonwealth of Virginia, in accordance with Rules of the American Arbitration Association in effect at the time the arbitration is initiated, and judgment upon the award rendered by the arbitrators may be entered in any court jurisdiction thereof, Company agrees to submit to the exclusive jurisdiction of the State and Federal courts in the Commonwealth of Virginia.

11.  _Miscellaneous_.  The parties to this Agreement are independent contractors with respect to each other, and nothing in this Agreement shall create any association, partnership, joint venture or agency relationship between the parties.  

### End of EULA

**Comment by gitwjr** [comment]
Done. Close.

**Comment by jpiesing** [comment]
Looking at the original problem description, I don't think the original comment from @cta-source  is addressed.

We need to think about how people will download the DPCTF test suite once it's released.
We would then include the text from CTA Legal at an appropriate place in that workflow.
- It could be a web page that takes you to a second page of links to the content on the Akamai web server
- It could be in the dpctf-deploy project where before it downloads any test content.

@mbergman42 As the originator of this issue, what do you think?

**Comment by mbergman42** [comment]
@jpiesing, I agree. It only applies to content, however. Hosting content publicly on a site where they can pick and choose content isn't going to support this EULA requirement. 

> It could be a web page that takes you to a second page of links to the content on the Akamai web server

Agreed, except that the Akamai web server doesn't (easily) support any kind of access control. Each user authorization must be hand-curated by Will, via informal connections with another Akamai department that isn't formally responsible for the task. So we're stuck with a public site that makes it easy to bypass the EULA.

> It could be in the dpctf-deploy project where before it downloads any test content. 

This makes sense, although we're still not home and dry. We still have the content exposed to public view if someone finds the link, and it will be quite easy to publicly share a link to our "protected" content once one person has installed the test suite. 

I agree with reopening this issue. We should discuss options.

**Comment by jpiesing** [comment]
Why does it matter if there's a way to bypass the EULA?
Look at Big Buck Bunny for example, the download page, https://peach.blender.org/download/, includes the license but someone who was sufficiently determined could share the URLs from the following page with anyone.

**Comment by mbergman42** [comment]
Good question. I'm meeting with Legal on that question in an hour. I'll report back. (Basically--if we don't have a 'hard' click-through that fully blocks access to the content without agreement, what is still acceptable that WAVE can implement?)

**Comment by mbergman42** [comment]
Per CTA Legal: We can post the EULA information on a website that gates access to the test material. This can be in DPCTF Deploy. It also should be on a web site that provides access to the test material, and when we need to give people access to the vector files, it should be via that web page, not the FTP-like dash-if.akamized.net/WAVE/vectors direct link. The existence of the public link is not optimal but because everything is protected by CCA3 and CCA4, it's OK if someone discovers it. The EULA protects CTA when CTA is pointing at the material.

Bottom line: 
1) We need to establish the EULA as a click-through requirement in Deploy, if possible.
2) If we want to give people direct access to the vectors under the Test Suite, we need a web page front-end that includes the EULA.

**Comment by jpiesing** [comment]
@cta-source Mike, perhaps you could open an issue in the deploy project that the EULA needs to be shown and agreement obtained before the test content is downloaded.
We do give people direct access to the test vectors from https://cta-wave.github.io/Test-Content/ . I guess that page needs to be revised to include the EULA, either before the page is shown or between selecting an MPD and the MPD being delivered. It presumably needs the whole cookie thing to save the results of accepting the EULA.

**Comment by mbergman42** [comment]
Done, see [Issue 32: Deploy must show an EULA](https://github.com/cta-wave/dpctf-deploy/issues/32). I've also created a EULA file and loaded it into DPCTF-deploy.

**Comment by gitwjr** [comment]
@rbouqueau to add an explicit EULA copyright statement on the content page and review with @mbergman.

**Comment by rbouqueau** [comment]
Deployed on https://cta-wave.github.io/Test-Content/. Is it ok?

**Comment by mbergman42** [comment]
This looks great, thanks. It appears to be fully functional. From my perspective if we can merge the PR, we can close this issue and the related *[cta-wave/dpctf-deploy] Deploy must show an EULA  (Issue #32)*.

Mike Bergman
Consumer Technology Association, producer of CES®
m: +1(609) 865-4402

**Comment by gitwjr** [comment]
Closed. See Deploy Issue #32 and PR #36. When starting the docker container, the startup process will fail unless a startup flag is provided.
